core/src/main/java/org/apache/iceberg/rest/requests/RegisterTableRequest.java

```diff
@@ -28,6 +28,8 @@ public interface RegisterTableRequest extends RESTRequest {
 
   String metadataLocation();
 
+  Boolean overwrite();
```

Comment:
nit: `boolean` primitive type?

Reply:
Use Boolean here as in #12239, we add `overwrite` as an optional field. I feel Boolean is the best way to convey the idea. Use primitive boolean default to false means we will always serialize/deserialize them in RequestParser. Currently I am only include this field if it's explicitly set